example_client/grpc_stateful_client.py

```diff
@@ -32,154 +32,250 @@ def printDebug(msg):
-    printDebug('\tSequence id: {}'.format(sequence_id))
+    if 'sequence_id' not in result.outputs:
+        print("ERROR: Missing sequence_id in model output")
+        print("Available outputs:")
+        for Y in result.outputs:
+            print(Y)
+        exit(1)
 
-    meanErrSum = 0.0
+def main():
+    args = ParseArguments()
+    global debug_mode
+    debug_mode = int(args.get('debug'))
+
+    channel = grpc.insecure_channel("{}:{}".format(args['grpc_address'],args['grpc_port']))
+    stub = prediction_service_pb2_grpc.PredictionServiceStub(channel)
+    processing_times = np.zeros((0),int)
+    cw_l = int(args.get('cw_l'))
+    cw_r = int(args.get('cw_r'))
+    print('\tContext window left width cw_l: {}'.format(cw_l))
+    print('\tContext window right width cw_r: {}'.format(cw_r))
+    print('Start processing:')
+    print('\tModel name: {}'.format(args.get('model_name')))
+
+    initial_ark_reader, ark_scores, input_names, output_names, numberOfKeys, inputArrays, referenceArrays = PrepareProcessingData(args)
+
+    SEQUENCE_START = 1
+    SEQUENCE_END = 2
+    sequence_id = 2
+    meanErrGlobal = 0.0
+
+    for key, obj in initial_ark_reader:
+        batch_size = obj.shape[0]
+        print('\n\tInput name: {}'.format(key))
+        printDebug('\tInput in shape: {}'.format(obj.shape))
+        printDebug('\tInput batch size: {}'.format(batch_size))
+        printDebug('\tSequence id: {}'.format(sequence_id))
+
+        meanErrSum = 0.0
+        score_index = (cw_l + cw_r) * -1
+
+        for x in range(0, batch_size + cw_l + cw_r):
+            printDebug('\tExecution: {}\n'.format(x))
+            request = predict_pb2.PredictRequest()
+            request.model_spec.name = args.get('model_name')
+
+            # Set input data index
+            input_index = x
+            # Input for context window
+            if x < cw_l:
+                # Left context window
+                input_index = 0
+            elif x >= cw_l and x < batch_size + cw_l:
+                # Middle window
+                input_index = x-cw_l
+            else:
+                # Right context window
+                input_index = batch_size - 1
+
+            # Setting request input
+            for input_name in input_names:
+                inputSubArray = inputArrays[input_name]
+                inputData = inputSubArray[key][input_index]
+                request.inputs[input_name].CopyFrom(make_tensor_proto(inputData, shape=inputData.shape))
+
+            # Add sequence start
+            if x == 0:
+                request.inputs['sequence_control_input'].CopyFrom(make_tensor_proto(SEQUENCE_START, dtype="uint32"))
+        
+            # Set sequence id
+            request.inputs['sequence_id'].CopyFrom(make_tensor_proto(sequence_id, dtype="uint64"))
 
-    samples_limit = int(args.get('samples'))
-    for x in range(0, batch_size):
+            # Add sequence end
+            if x == batch_size + cw_l + cw_r - 1:
```

Comment:
When there's only one frame in the sequence it can't be handled with this condition.

Reply:
If we don't want to support single request sequences, then you could throw error when user provides such sequence.